### PR TITLE
Add RGB to Hexadecimal mapping function

### LINKED ISSUE #9: Convert RGB values to HEX

### PR DESCRIPTION
**Pull request by ajisakson**

### CODE REVIEW

File: src/utils/extract.js [hunk 1]
```diff
@@ -63,5 +63,22 @@ export const extract = {
...
+            redHex = '0' + redHex;
+        }
+        if (greenHex.length === 1) {
+            greenHex = '0' + greenHex;
+        }
+        if (blueHex.length === 1) {
+            blueHex = '0' + blueHex;
+        }
+
+        return '#' + redHex + greenHex + blueHex;
```
Comment (jakecjones) [inline src/utils/extract.js:82]: Could we use string interpolation here? return \`#${redHex}${greenHex}${blueHex}\`;

File: src/utils/extract.js [hunk 2]
```diff
@@ -63,5 +63,22 @@ export const extract = {
         });
   
       return clusteredAverages;
+    },
+    rgbToHexadecimal: (r,g,b) => {
```
Comment (jakecjones) [inline src/utils/extract.js:67]: #nitpick but could we change these parameters to `red, green, blue`?

Comment (jakecjones) [inline src/utils/extract.js]: #nitpick but could we change these parameters to `red, green, blue`?

File: src/views/Home.vue [hunk 3]
```diff
@@ -142,7 +142,10 @@ export default {
         });
         
         for (var colorIndex = 0; colorIndex < this.colors; colorIndex++) {
-          outputs.push(this.clusteredColors[colorIndex][0])
+          let newColor = this.clusteredColors[colorIndex][0];
+          newColor.hex = extract.rgbToHexadecimal(newColor.red, newColor.green, newColor.blue);
```
Comment (AndreBClark) [inline src/views/Home.vue]: `newColors` seems a bit too ambiguous, does this variable hold the value of the combined averages for all the pixel of an image? would `combinedColor` be more descriptive?

File: src/utils/extract.js (hunk 2 above)
Comment (AndreBClark) [reply src/utils/extract.js]: I agree with this. it's a bit of an edge case as RGB is pretty self-explanatory, however, I think we should use descriptive variable names as much as possible for maintainability and readability.

File: src/utils/extract.js (hunk 1 above)
Comment (AndreBClark) [reply src/utils/extract.js:82]: concatenation and template literals are interchangeable here. this feels more readable with concatenation, which might be the only case I would prefer concatenation over template literals. as this is mostly variables and barely any strings, and doesn't save any space. could go either way honestly.

**Review by AndreBClark — CHANGES_REQUESTED** [review]
rename the r, g, b variables and the newColor variable to a more descriptive name and I would approve this, if everyone agrees.

**Comment by jakecjones** [comment]
@AndreBClark i agree ✔️

File: src/utils/extract.js (hunk 1 above)
Comment (jakecjones) [reply src/utils/extract.js:82]: I'd buy that. Readable is always better.

**Pull request merged**